packages/ionic/src/commands/capacitor/base.ts

```diff
@@ -0,0 +1,47 @@
+import { CommandInstanceInfo, IShellRunOptions } from '@ionic/cli-utils';
+import { Command } from '@ionic/cli-utils/lib/command';
+import { FatalException } from '@ionic/cli-utils/lib/errors';
+import { runCommand } from '@ionic/cli-utils/lib/executor';
+
+export abstract class CapacitorCommand extends Command {
+  async checkCapacitor(runinfo: CommandInstanceInfo) {
+    const project = await this.env.project.load();
+
+    if (project.integrations.capacitor && project.integrations.capacitor.enabled === false) {
+      return;
+    }
+    if (!project.integrations.capacitor) {
+      await runCommand(runinfo, ['integrations', 'enable', 'capacitor']);
+    }
+  }
+
+  async preRunChecks(runinfo: CommandInstanceInfo) {
+    runinfo;
+  }
+
+  async runCapacitor(argList: string[], { fatalOnNotFound = false, truncateErrorOutput = 5000, ...options }: IShellRunOptions = {}): Promise<void> {
+    const { ERROR_SHELL_COMMAND_NOT_FOUND } = await import('@ionic/cli-utils/lib/shell');
+    try {
+      this.env.close();
+      await this.env.shell.run('npx', ['capacitor'].concat(argList), { fatalOnNotFound, truncateErrorOutput, stdio: 'inherit', ...options });
+      this.env.open();
+    } catch (e) {
+      this.env.open();
+      if (e === ERROR_SHELL_COMMAND_NOT_FOUND) {
+        throw new FatalException(
+          `Unable to run npx, the new command execution system in NPM.
+          This means your Node version is too old. Capacitor requires Node 8.6.0 or greater,
```

Comment:
Is this node requirement because of the Capacitor CLI codebase, or `npx`?

Reply:
Without npx as above it won't be a requirement (but yes, somewhere in Node 5 npx was added as a command without having to install something else)